Dopamine/Dopamine.swift

```diff
@@ -0,0 +1,279 @@
+//
+//  Dopamine.swift
+//
+//  Copyright (c) 2015 Dopamine Labs
+//
+//  Released under The MIT License (MIT)
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in all
+//  copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+//  SOFTWARE.
+
+import Foundation
+import Alamofire
+import CryptoSwift
+import SwiftyJSON
+
+/// Generic settings for connecting to Dopamine
+typealias Settings = (appID: String, apiKey: String, token: String, versionID: String, pairings: ActionPairings)
+/// Reinforcers are rewards or feedback for the end user which are primarily identified by name
+typealias Reinforcer = (name: String, type: FunctionType, constraint: [String], objective: [String] )
+/// Actions have a 1:n pairing with possible rewards and feedback (Reinforcer)
+typealias ActionPairing = (name: String, reinforcers: [Reinforcer])
+/// A map of names to action pairings
+typealias PairingDictionary = [String:ActionPairing]
+/// Convenience type for the request payload
+typealias APIPayload = [String:AnyObject]
+/// Closure of the form {(statusCode, functionName) -> Void} that is provided a callback on reinforcements
+public typealias ReinforcementCallback = ((Int, String) -> Void)
+
+/**
+    The API is namespaced by suffixes. This enum defines the appropriate sufixes to be added to the
+    end of a URL
+*/
+enum APIEndpoint : String {
```

Comment:
@DavidAntaramian Swift style would be to put the colon right after the enum name `enum APIEndpoint: String {`